src/bot/mod.rs

```diff
@@ -27,6 +27,7 @@ pub(crate) const TELOXIDE_PROXY: &str = "TELOXIDE_PROXY";
 #[derive(Debug, Clone)]
 pub struct Bot {
     token: Arc<str>,
+    api_url: Option<Arc<str>>,
```

Comment:
Maybe a wrapper around `Arc<str>` with the default implementation would be better?

Reply:
For one place, I don't think wrapper would make the code better...